cmake/helpers.cmake

```diff
@@ -0,0 +1,297 @@
+#
+# Copyright (c) 2023 ZettaScale Technology
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+# which is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+#
+# Contributors:
+#   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+#
+
+#
+# DO NOT add include_guard() to this file
+# It have to be reincluded multiple times on each 'include_project' call to restore
+# functions and macros definitions
+#
+
+#
+# Show VARIABLE = value on configuration stage
+#
+macro(status_print var)
+	message(STATUS "${var} = ${${var}}")
+endmacro()
+
+#
+# Declare cache variable and print VARIABLE = value on configuration stage
+#
+function(declare_cache_var var default_value type docstring)
+	set(${var} ${default_value} CACHE ${type} ${docstring})
```

Comment:
For this use cmake's "option" instead. And yes, option has default value (and "reset to default" operation that is used in IDEs, for example, QtCreator) which is comfortable to use in cmake's code.

Reply:
Option is limited with boolean values. As far as I understand there is no difference between "option" and set(... CACHE BOOL ...)